### PR TITLE
FAQ.rst: Add ToC

### PR DESCRIPTION
**Pull request by Warewolf96**
Added Table of contents in FAQ.rst

Closes https://github.com/coala/documentation/issues/285

### CODE REVIEW

File: Help/FAQ.rst [hunk 1]
```diff
@@ -4,6 +4,19 @@ Frequently Asked Questions
 This is a list of frequently asked questions, aiming to answer any possible
 questions by newcomers or even contributors.
 
+Table of Contents
+-----------------
+- What does coala do (for me)?_.
+- coala in my Continuous Integration_.
```
Comment (pratyushprakash) [inline Help/FAQ.rst]: Could you change this to `Can I Use coala in my Continuous Integration?`

Comment (Warewolf96) [inline Help/FAQ.rst]: Done.. :)

File: Help/FAQ.rst [hunk 2]
```diff
@@ -4,6 +4,19 @@ Frequently Asked Questions
 This is a list of frequently asked questions, aiming to answer any possible
 questions by newcomers or even contributors.
 
+Table of Contents
+-----------------
+- What does coala do (for me)?_.
+- Can I Use coala in my Continuous Integration?_.
```
Comment (NiklasMM) [inline Help/FAQ.rst]: `Use` -> `use`

File: Help/FAQ.rst [hunk 3]
```diff
@@ -4,6 +4,19 @@ Frequently Asked Questions
...
+- Why did you choose the name?_.
+- Is there corporate backing behind coala? What are your intentions?_.
+- What sort of analysis does coala do? What languages are supported?_.
+- How do I get started with coala?_.
+- How do I get in touch with the coala team?_.
+- Installation is Failing! Help!_.
```
Comment (NiklasMM) [inline Help/FAQ.rst]: `Failing` -> `failing`

File: Help/FAQ.rst [hunk 4]
```diff
@@ -18,19 +31,22 @@ In short coala does two things:
   everything but the actual analyzer routine. Write just the parameters
   your custom analysis needs and the analysis part, we'll take care of the rest
 
+.. _Can I Use coala in my Continuous Integration?:
```
Comment (NiklasMM) [inline Help/FAQ.rst]: `Use` -> `use`

File: Help/FAQ.rst [hunk 5]
```diff
@@ -76,6 +95,7 @@ and will try to respond to any question in a matter of minutes.
 However, for a full list of how to get in touch with us, consult
 :doc:`this link <Getting_In_Touch>`.
 
+.. _Installation is Failing! Help!:
```
Comment (NiklasMM) [inline Help/FAQ.rst]: `Failing` -> `failing`

Comment (Warewolf96) [inline Help/FAQ.rst]: Done... :)

**Comment by NiklasMM** [comment]
ack cd7609d

**Comment by Warewolf96** [comment]
@sils can u please merge this?

**Comment by sils** [comment]
Uhm IIRC you can generate the ToC with restructured text, are we sure we have to hardcode it here?

**Comment by Warewolf96** [comment]
@sils please elaborate.. actually i don't have much knowledge about rst so i studied about it and this was the first method i could find to solve the issue

**Comment by NiklasMM** [comment]
I also don't know much about this, but what sils means is, that we probably don't have to keep the TOC hardcoded like this, but have it be dynamically generated. This way, if new sections are added, the TOC is automatically up-to-date. Maybe look at [this](http://docutils.sourceforge.net/docs/ref/rst/directives.html#table-of-contents) to start finding out how this would be done.

**Comment by Mixih** [comment]
unack cd7609d

**Comment by Mixih** [comment]
autogen advised

**Comment by Warewolf96** [comment]
ohk i'm on it..

**Comment by Warewolf96** [comment]
@sils I think this should do...

**Comment by sils** [comment]
ack https://github.com/coala/documentation/pull/400/commits/014b26aa6438a41f7d12f278067142e0e34f39cc?short_path=1962993#diff-196299309d95d7684abefaeded3842a6 :)

**Comment by sils** [comment]
needs rebase or maybe @Mariatta cherry picks it ;)

**Comment by Warewolf96** [comment]
@sils i have to do anything else?

**Comment by Warewolf96** [comment]
So, should i make a new branch for this?

**Comment by Makman2** [comment]
@Warewolf96 no please read in our newcomer guide regarding rebasing^^

**Comment by sims1253** [comment]
this still needs a rebase

**Comment by Warewolf96** [comment]
this PR had some problems I'm going to open a new one which will be easy to rebase and merge

**Pull request closed without merging**